src/hdf5plugin/_filters.py

```diff
@@ -500,6 +503,75 @@ def __pack_float64(error: float) -> tuple:
         return high, low
 
 
+class SZ3(_FilterRefClass):
+    """``h5py.Group.create_dataset``'s compression arguments for using SZ3 filter.
+
+    void usage()
+    {
+            printf("Usage: print_h5repack_args <options>\n");
+            printf("Options:\n");
+            printf("	-M <error bound mode> : 10 options as follows. \n");
+            printf("		ABS (absolute error bound)\n");
+            printf("		REL (value range based error bound, so a.k.a., VR_REL)\n");
+            printf("		PSNR (peak signal-to-noise ratio)\n");
+            printf("		NORM2 (norm2)\n");
+            printf("	-A <absolute error bound>: specifying absolute error bound\n");
+            printf("	-R <value_range based relative error bound>: specifying relative error bound\n");
+            printf("	-N <norm2>: specifying norm2 error bound\n");
+            printf("	-S <PSNR>: specifying PSNR\n");
+            printf("* examples: \n");
+            printf("	print_h5repack_args -M ABS -A 1E-3 (output: -f UD=32024,0,9,0,1062232653,3539053052,0,0,0,0,0,0)\n");
+            printf("	print_h5repack_args -M REL -R 1E-4 (output: -f UD=32024,0,9,1,0,0,1058682594,3944497965,0,0,0,0)\n");
+            exit(0);
+    }
+
+
+    For more details about the compressor `SZ <https://szcompressor.org/>`_.
+    """
+    filter_name = "sz3"
+    filter_id = SZ3_ID
+
+    def __init__(self, absolute=None, relative=None, norm2=None, peak_signal_to_noise_ratio=None):
```

Comment:
For a first version with limited features, I would limit the arguments:
```suggestion
    def __init__(self, absolute=None):
        relative = None
        norm2 = None
        peak_signal_to_noise_ratio = None
```

Reply:
I would not limit the arguments, but only document those used. In any case I am going to try to see how many are working.